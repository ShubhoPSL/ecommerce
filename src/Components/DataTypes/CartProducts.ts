export type CartProducts={
  id: number;
  title: string;
  price: number;
  image: string;
  count: number;
}