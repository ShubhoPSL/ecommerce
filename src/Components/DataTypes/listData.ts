export type listData = {
    name: string;
    route: string;
  };