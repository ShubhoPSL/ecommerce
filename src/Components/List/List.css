.nav-li {
  @apply md:mx-5 lg:mx-10 text-xl lg:text-2xl;
}
.cart-div {
  @apply inline-block py-2 w-[40px] h-[40px] relative;
}
.cart-para {
  @apply absolute top-0 left-[28px] my-1 bg-yellow-400 rounded-full text-xs font-extrabold p-[2px];
}
