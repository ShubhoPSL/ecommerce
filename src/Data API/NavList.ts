export const navList = [
    {
      name: "Home",
      route: "home",
    },
    {
      name: "Products",
      route: "products",
    },
    {
      name: "Contact Us",
      route: "contact-us",
    },
    {
      name: "Cart",
      route: "cart",
    },
  ];