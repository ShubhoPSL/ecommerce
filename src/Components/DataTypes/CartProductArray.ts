import { CartProducts } from "./CartProducts";

export type cartProductArray=CartProducts[]