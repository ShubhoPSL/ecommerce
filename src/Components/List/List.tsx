import React from "react";
import { useRecoilValue } from "recoil";
import { Link } from "react-router-dom";
import { listData } from "../DataTypes/listData";
import { TotalItemsInCart } from "../States/TotalItemsInCart";
import { CartIcon } from "../Icons/CartIcon";
const List = ({ data }: { data: listData }) => {
  const totalItems = useRecoilValue(TotalItemsInCart);
  return (
    <li className="nav-li">
      <Link to={data.route}>
        {data.name === "Cart" ? (
          <div className="cart-div">
            <CartIcon />
            <p className="cart-para">{totalItems}</p>
          </div>
        ) : (
          data.name
        )}
      </Link>
    </li>
  );
};

export default List;
