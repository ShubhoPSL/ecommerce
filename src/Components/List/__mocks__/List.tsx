import React from "react";
import { listData } from "../../DataTypes/listData";

const List = ({ data }: { data: listData }) => {
  return <div>List - {data.name}</div>;
};

export default List;
