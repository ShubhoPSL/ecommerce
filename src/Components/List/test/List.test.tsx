import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { RecoilRoot } from "recoil";
import { navList } from "../../../Data API/NavList";
import { listData } from "../../DataTypes/listData";
import List from "../List";
import renderer from "react-test-renderer";

jest.mock("../../Icons/CartIcon");
jest.mock("../../States/CartItems");
describe("Testing List component", () => {
  const NewList: React.FC<{ data: listData }> = ({
    data,
  }: {
    data: listData;
  }) => {
    return (
      <RecoilRoot>
        <BrowserRouter>
          <List data={data} />
        </BrowserRouter>
      </RecoilRoot>
    );
  };
  it("should have Home item", () => {
    render(<NewList data={navList[0]} />);
    const element = screen.getByRole("link", { name: "Home" });
    expect(element).toBeInTheDocument();
  });
  it("should have Products item", () => {
    render(<NewList data={navList[1]} />);
    const element = screen.getByRole("link", { name: "Products" });
    expect(element).toBeInTheDocument();
  });
  it("should have Contact Us item", () => {
    render(<NewList data={navList[2]} />);
    const element = screen.getByRole("link", { name: "Contact Us" });
    expect(element).toBeInTheDocument();
  });
  it("should have Cart item", () => {
    render(<NewList data={navList[3]} />);
    const element = screen.getByRole("link", { name: /Cart/ });
    expect(element).toBeInTheDocument();
  });
  it("should have Cart item with 2 item", () => {
    render(<NewList data={navList[3]} />);
    const element = screen.getByText(/2/);
    expect(element).toBeInTheDocument();
  });
  it("should have Cart item with 0 item", () => {
    jest.requireActual("../../States/CartItems");
    render(<NewList data={navList[3]} />);
    const element = screen.getByText(/0/);
    expect(element).toBeInTheDocument();
  });
  it("should match with snapshot of List module", () => {
    const tree = renderer.create(<NewList data={navList[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
