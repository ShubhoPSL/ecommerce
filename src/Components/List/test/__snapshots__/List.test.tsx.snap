// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List component should match with snapshot of List module 1`] = `
<li
  className="nav-li"
>
  <a
    href="/home"
    onClick={[Function]}
  >
    Home
  </a>
</li>
`;
